### PR TITLE
Pull (most of) Footer from `frontpage` into the DS

### PR DESCRIPTION
**Pull request by smithambera**
Fixes https://linear.app/chromaui/issue/CH-708/pull-common-page-footer-from-frontpage-into-ds

## Description

Pulling the Footer component from https://github.com/storybookjs/frontpage into the DS in preparation for future work.

### Of Note

- Following [this proposal](https://www.notion.so/Styling-solutions-across-design-system-and-its-consumers-aba73999c0114448ad258ae4f0a24010) and branching off of the branch converting the DS to Emotion.
- I've left off the email subscription for the time being and created [an issue](https://linear.app/chromaui/issue/CH-801/pull-confirmed-mailing-list-from-frontpage-into-ds) to pull that into the DS as well.

### QA instructions

Confirm the styles (mostly) match [the real thing](https://storybook.js.org/).

<details>
  <summary>📦 Published PR as canary version: <code>6.4.2-canary.314.b40307e.0</code></summary>
  <br />
  
  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @storybook/design-system@6.4.2-canary.314.b40307e.0
  # or 
  yarn add @storybook/design-system@6.4.2-canary.314.b40307e.0
  ```
</details>

### CODE REVIEW

File: src/components/Subheading.tsx [hunk 1]
```diff
@@ -20,11 +20,12 @@ const StyledSubheading = styled('span', { shouldForwardProp: (prop) => !['muted'
...
-export const Subheading: FunctionComponent<Props> = ({ muted = false, ...props }: Props) => (
-  <StyledSubheading muted={muted} {...props} />
-);
+export const Subheading: FunctionComponent<SubheadingProps> = ({
+  muted = false,
+  ...props
+}: SubheadingProps) => <StyledSubheading muted={muted} {...props} />;
 
-interface Props {
+export interface SubheadingProps {
```
Comment (smithambera) [inline src/components/Subheading.tsx:28]: I did this cuz typescript was mad about me styling `Subheading` in another file.

File: src/components/marketing/PageFooter/PageFooter.styles.ts [hunk 2]
```diff
@@ -0,0 +1,265 @@
+import { styled } from '@storybook/theming';
```
Comment (winkerVSbecks) [inline src/components/marketing/PageFooter/PageFooter.styles.ts:1]: `*.styles.ts` is a cool pattern

File: src/components/marketing/PageFooter/PageFooter.tsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+import ChromaticLogoSVG from '../../../images/logos/chromatic.svg';
+// @ts-ignore
+import CircleCILogoSVG from '../../../images/logos/circleci.svg';
+
+export const PageFooter = ({
+  coreFrameworks,
+  tutorialsLink,
+  storybookLogoLink,
+  navLinks,
+  communityLinks,
```
Comment (winkerVSbecks) [inline src/components/marketing/PageFooter/PageFooter.tsx:42]: These will be the same everywhere. So I reckon we can just keep these internal and not require a prop for it.

File: src/components/marketing/PageFooter/PageFooter.tsx [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+import RepoSVG from '../../../images/repo.svg';
+// @ts-ignore
+import NetlifyLogoSVG from '../../../images/logos/netlify.svg';
+// @ts-ignore
+import ChromaticLogoSVG from '../../../images/logos/chromatic.svg';
+// @ts-ignore
+import CircleCILogoSVG from '../../../images/logos/circleci.svg';
+
+export const PageFooter = ({
+  coreFrameworks,
```
Comment (winkerVSbecks) [inline src/components/marketing/PageFooter/PageFooter.tsx:38]: These will also be the same everywhere, I would suggest moving it to an internal config

**Review by winkerVSbecks — APPROVED** [review]

File: src/images/direction.svg [hunk 5]
```diff
@@ -0,0 +1 @@
...rg/1999/xlink" width="48" height="48" version="1.1" viewBox="0 0 48 48"><title>illustration/direction</title><desc>Created with Sketch.</desc><g id="illustration/direction" fill="none" fill-rule="evenodd" stroke="none" stroke-width="1"><path id="Combined-Shape" fill="#FFD476" d="M23.4917015,33.6030641 L2.93840258,31.4321033 C2.38917316,31.3740904 1.99096346,30.8818233 2.04897631,30.3325939 C2.0747515,30.0885705 2.18934861,29.8625419 2.37095722,29.6975265 L34.2609105,0.721285325 C34.6696614,0.349881049 35.3021022,0.38015648 35.6735064,0.788907393 C35.9232621,1.06377731 36.0001133,1.45442096 35.8730901,1.80341447 L24.5364357,32.9506164 C24.3793473,33.3822133 23.9484565,33.6513092 23.4917015,33.6030641 Z"/><path id="Combined-Shape-Copy" fill="#FFC445" d="M24.3163597,33.2881029 C24.0306575,33.0138462 23.9337246,32.5968232 24.069176,32.2246735 L35.091923,1.9399251 C35.2266075,1.56988243 35.5659249,1.31333613 35.9586669,1.28460955 C36.5094802,1.24432106 36.9886628,1.65818318 37.0289513,2.20899647 L40.2437557,46.1609256 C40.2644355,46.4436546 40.1641446,46.7218752 39.9678293,46.9263833 C39.5853672,47.3248067 38.9523344,47.3377458 38.5539111,46.9552837 L24.3163597,33.2881029 Z"/></g></svg>
```
Comment (domyen) [inline src/images/direction.svg:1]: Perhaps it's time to move all [these images/illustrations](https://59c59bd0183bd100364e1d57-qylvkuphmr.chromatic.com/?path=/story/images--colored-icons) from Chromatic into SDS. They're also duplicated on the Storybook website.  

What do you think?

File: src/components/marketing/PageFooter/PageFooter.styles.ts (hunk 2 above)
Comment (domyen) [reply src/components/marketing/PageFooter/PageFooter.styles.ts:1]: Very CSS-like! 

Cool that the style file follows the same naming convention as the component.

File: src/images/direction.svg (hunk 5 above)
Comment (smithambera) [reply src/images/direction.svg:1]: I'll make an issue to do just that!

Comment (smithambera) [reply src/images/direction.svg:1]: https://linear.app/chromaui/issue/CH-806/pull-colored-icons-from-chromatic-frontpage-into-ds 🎉

**Review by kylegach — APPROVED** [review]
Looks good!

**Pull request merged**